lib/Analysis/NoiseAnalysis/BFV/NoiseByVarianceCoeffModel.h

```diff
@@ -0,0 +1,76 @@
+#ifndef INCLUDE_ANALYSIS_NOISEANALYSIS_BFV_NOISEBYVARIANCECOEFFMODEL_H_
+#define INCLUDE_ANALYSIS_NOISEANALYSIS_BFV_NOISEBYVARIANCECOEFFMODEL_H_
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstdint>
+#include <iomanip>
+#include <optional>
+#include <string>
+
+#include "lib/Analysis/NoiseAnalysis/NoiseWithDegree.h"
+#include "lib/Parameters/BGV/Params.h"
+
+namespace mlir {
+namespace heir {
+namespace bfv {
+
+// coefficient embedding noise model using variance
+class NoiseByVarianceCoeffModel {
+ public:
+  // NoiseState stores the variance var for the one coefficient of
+  // the error 'e', assuming coefficients are IID.
+  // Note that BMCM23 tracks the invariant noise t/q(c0 + c1 * s).
+  // We adapt it to track the 'e' in [(q/t)m + e]_q for convenience
+  // and consistency with the other models.
+  //
+  // The main contribution of BMCM23 is that it considers the dependence between
+  // random variables caused by 's', so the max degree T in 's^T' of the
+  // invariant noise needs to be tracked.
+  //
+  // MP24/CCH+23 states that for two polynomial multipication, the variance of
```

Comment:
Link to these two papers. I see them referred to in the BGV models as well. Maybe we should have a README under NoiseAnalysis that gives citations for all the papers and then we can refer to the acronyms in code comments (with the understanding that `grep MP24` will find the README)